x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryFolder.java

```diff
@@ -455,6 +456,7 @@ static EsQueryExec fold(AggregateExec a, EsQueryExec exec) {
                 // literal
                 if (target.foldable()) {
                     queryC = queryC.addColumn(ne.toAttribute());
+                    literalId = id;
```

Comment:
Maybe it's better to construct a `literalId` which concatenates all the ids.

Reply:
why? we only need one of them no matter which

Reply:
Sorry, I was wrong, It's fine as is, I just think that you can use directly `id`, no need of a new variable.

Reply:
id is set and used inside the for loop, and it is changing with the for loop, that's why I needed literalId

Reply:
I can move this `String id` outside the for loop and get rid of literalId, that's true. I'll check the failures, and do this at the same time.